Added tests for FruitShop classes

DanaKhromenko opened the pull request.

DanaKhromenko commented:
- added "@AfterAll" equivalent method where rewrite input file to default value;
- removed extra separation;
- moved '+' symbols to the new line.

DanaKhromenko commented:
Replaced invalid path value with empty path

RomanDubovskyi commented on src/test/java/core/basesyntax/FileManagerCsvImplTest.java:
```diff
@@ -0,0 +1,60 @@
+                + "b,grape,50\n"
+                + "b,melon,30";
+        // Can't import jupiter because of JUnit4
+        // So can't use AssertThrows also
```
ти можеш ставити 5 версію за бажанням 🙂

DanaKhromenko replied:
Но ведь в условии написано:
"Don’t use any other version of JUnit
Use JUnit4 that is already present in your pom.xml."...

RomanDubovskyi replied:
yep, my bad тут, в декого виникали проблеми з код кавереджем в 5 версії

RomanDubovskyi commented on src/test/java/core/basesyntax/FileWriter.java:
```diff
@@ -0,0 +1,36 @@
+
+public class FileWriter {
+    public static void writeDataToFile(String filePath, String data) {
+        File file = new File(filePath);
+        if (file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                throw new RuntimeException("Can't create file " + filePath, e);
+            }
+        }
+        try (BufferedWriter bufferedWriter = new BufferedWriter(
+                new java.io.FileWriter(filePath, false))) {
+            bufferedWriter.write(data);
```
а чому якщо file.exists() == true ми викликаємо file.createNewFile ?
```suggestion
        try (BufferedWriter bufferedWriter = new BufferedWriter(
                new java.io.FileWriter(filePath, false))) {
            bufferedWriter.write(data);
```

DanaKhromenko replied:
Оу, там должно было быть !file.exists()

RomanDubovskyi commented on src/test/java/core/basesyntax/ShopOperationsDaoCsvImplTest.java:
```diff
@@ -0,0 +1,114 @@
+                + "b,melon,30");
+        Assert.assertFalse(shopOperationsDao.validate());
+
+        // AfterAll doesn't work
```
якщо ти хочеш щоб renewInputFile скидувався після кожного тесту то тобі потрібен AfterEach / After в залежності від весрії junit а не AfterAll / AfterClass

RomanDubovskyi commented on src/test/java/core/basesyntax/FruitDataValidatorCsvImplTest.java:
```diff
@@ -0,0 +1,45 @@
+import validators.FruitDataValidatorCsvImpl;
+
+public class FruitDataValidatorCsvImplTest {
+    private static final FruitDataValidator VALIDATOR = new FruitDataValidatorCsvImpl();
```
```suggestion
    private static final FruitDataValidator validator = new FruitDataValidatorCsvImpl();
```
якщо ми не робимо клас deeply immutable то неймінг повинен бути як простого класу (String має такий неймінг так як це immutable class)

RomanDubovskyi commented on src/test/java/core/basesyntax/ShopOperationsDaoCsvImplTest.java:
```diff
@@ -0,0 +1,114 @@
+        String actual = getDataFromFile();
+        Assert.assertEquals(expected, actual);
+
+        FileWriter.writeDataToFile(INPUT_FILE_NAME, "type,fruit,quantity\n"
```
цей тест по логіці роботи не відрізняється від першого крім вхідних даних, думаю такі кейси можна скіпнути

RomanDubovskyi commented on src/test/java/core/basesyntax/ShopOperationsDaoCsvImplTest.java:
```diff
@@ -0,0 +1,114 @@
+        FileWriter.renewInputFile(INPUT_FILE_NAME);
+    }
+
+    private String getDataFromFile() {
```
код в методі дублюється, можна зробити так щоб він на вхід приймав шлях до файлу і реюзнути його в тест класах, де потрібно

DanaKhromenko replied:
Не уверена, что поняла вторую половину фразы. Вынесла "FileWriter.renewInputFile(fileName);" в отдельный метод "renewFile" класса ShopOperationsDaoCsvImplTest.

RomanDubovskyi replied:
> Не уверена, что поняла вторую половину фразы. Вынесла "FileWriter.renewInputFile(fileName);" в отдельный метод "renewFile" класса ShopOperationsDaoCsvImplTest.

цей коментар стосувався методу getDataFromFile()
![image](https://user-images.githubusercontent.com/58385506/128018398-99f0751e-c55a-4af1-91aa-6a33b889fd8e.png)

DanaKhromenko commented:
- renamed variables to match the String-naming rules;
- removed file creating code in method writeDataToFile;
- removed similar test case;
- replaced FileWriter.renewInputFile(fileName) calls with class method renewFile.

RomanDubovskyi commented on src/test/java/core/basesyntax/ShopOperationsDaoCsvImplTest.java:
```diff
@@ -0,0 +1,96 @@
+import org.junit.Test;
+
+public class ShopOperationsDaoCsvImplTest {
+    private static final String inputFileName = "fruit-shop.csv";
```
```suggestion
    private static final String INPUT_FILE = "fruit-shop.csv";
```
String - deeply immutable 
https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

DanaKhromenko replied:
Так, смотри, то, что immutable, то капсом, типа Стринга и наших immutable классов. Всё остальное не immutable - не капсом, а как обычно.
Например,
private static final String str;
private static final NonImmutableCat cat;
private static final ImmutableCat CAT;
Верно?

RomanDubovskyi commented on src/test/java/core/basesyntax/FileWriter.java:
```diff
@@ -0,0 +1,27 @@
+        }
+    }
+
+    public static void renewInputFile(String filePath) {
```
сорс код на має бути залежний від тестів якщо потрібно додавати ютільний код для тестів він має бути в пакеті test а не src

RomanDubovskyi commented on src/test/java/core/basesyntax/ShopOperationsDaoCsvImplTest.java:
```diff
@@ -0,0 +1,96 @@
+        return stringBuilder.toString().trim();
+    }
+
+    private void renewFile(String fileName) {
```
створи тут метод: 
```
	@After
    public void teardown() {
       // your logic that should repeated after each method 
    }
```

DanaKhromenko commented:
- changed constants names;
- renamed FileWriter to FileAdministrator; 
- removed FileAdministrator to the new util package; 
- removed getDataFromFile-method from test classes to FileAdministrator;
- removed the call of renewing input file to teardown-method with annotation @After.
